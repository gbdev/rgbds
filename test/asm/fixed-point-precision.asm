f1 = 3.1
f2 = 5.2
pm = MUL(f1, f2)
pr = 16.12
	println "`3.1`:   {9.6f:f1} -> ${08x:f1}"
	println "`5.2`:   {9.6f:f2} -> ${08x:f2}"
	println "`MUL`:   {9.6f:pm} -> ${08x:pm}"
	println "`16.12`: {9.6f:pr} -> ${08x:pr}"

fl = 6.283185
	println "`6.283185`: {.6f:fl} -> ${08x:fl}"

fr = MUL(20.0, 0.32)
	println "32% of 20 = {f:fr} (~{.2f:fr}) (~~{.0f:fr})"

q8 = 1.25q8
q16 = 1.25Q16
q24 = 1.25q.24
	println "Q8 ${x:q8} Q16 ${x:q16} Q24 ${x:q24}"

qerr = 1.25q32
	println qerr
