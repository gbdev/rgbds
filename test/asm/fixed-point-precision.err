error: fixed-point-precision.asm(21):
    Fixed-point constant precision must be between 1 and 31
error: Assembly aborted (1 error)!
