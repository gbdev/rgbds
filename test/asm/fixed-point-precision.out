`3.1`:    3.100006 -> $0003199a
`5.2`:    5.199997 -> $00053333
`MUL`:   16.120026 -> $00101eba
`16.12`: 16.119995 -> $00101eb8
`6.283185`: 6.283188 -> $0006487f
32% of 20 = 6.40015 (~6.40) (~~6)
Q8 $140 Q16 $14000 Q24 $1400000
$14000
